tests/org.jboss.tools.cdi.bot.test/src/org/jboss/tools/cdi/bot/test/validation/cdi20/CDIValidatorTestCDI20.java

```diff
@@ -0,0 +1,59 @@
+/*******************************************************************************
+ * Copyright (c) 2019 Red Hat, Inc.
+ * Distributed under license by Red Hat, Inc. All rights reserved.
+ * This program is made available under the terms of the
+ * Eclipse Public License v1.0 which accompanies this distribution,
+ * and is available at http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributor:
+ *     Red Hat, Inc. - initial API and implementation
+ ******************************************************************************/
+package org.jboss.tools.cdi.bot.test.validation.cdi20;
+
+import java.util.Arrays;
+import java.util.Collection;
+
+import org.eclipse.reddeer.eclipse.ui.perspectives.JavaEEPerspective;
+import org.eclipse.reddeer.junit.annotation.RequirementRestriction;
+import org.eclipse.reddeer.junit.requirement.matcher.RequirementMatcher;
+import org.eclipse.reddeer.requirements.jre.JRERequirement.JRE;
+import org.eclipse.reddeer.requirements.openperspective.OpenPerspectiveRequirement.OpenPerspective;
+import org.eclipse.reddeer.requirements.server.ServerRequirementState;
+import org.jboss.ide.eclipse.as.reddeer.server.family.ServerMatcher;
+import org.jboss.ide.eclipse.as.reddeer.server.requirement.ServerRequirement.JBossServer;
+import org.jboss.tools.cdi.bot.test.validation.template.CDIValidatorTemplate;
+import org.junit.Before;
+
+/**
+*
+* @author zcervink@redhat.com
+*
+*/
+@JRE(cleanup=true)
+@JBossServer(state = ServerRequirementState.PRESENT, cleanup=false)
+@OpenPerspective(JavaEEPerspective.class)
+public class CDIValidatorTestCDI20 extends CDIValidatorTemplate{
+
+	@RequirementRestriction
+	public static Collection<RequirementMatcher> getRestrictionMatcher() {
+		if (isJavaLE8()) {
```

Comment:
As I said some time ago, would be awesome to also have test project running on Java 11 in case that test are running on machine with jdk 11. Also, I can see that this condition repeats in every class, do you dare to extract it somewhere else and just call in @requirementResctriction annotated method?

Reply:
Let's put it into some utility class, that's an easiest solution.

Reply:
After discussion with @psrna and @jkopriva this issue will be fixed after CDI PRs has been merged, issue created:

https://issues.jboss.org/browse/JBIDE-26750

Reply:
Great, I have updated the ticket also with a below mentioned server "parametrizetion" in pom.

Reply:
Thank you very much.